include/ada/href_aggregator.h

```diff
@@ -0,0 +1,82 @@
+/**
+ * @file href_aggreator.h
+ * @brief Definitions for the href_aggregator.
+ */
+#ifndef ADA_HREF_AGGREGATOR_H
+#define ADA_HREF_AGGREGATOR_H
+
+#include "ada/encoding_type.h"
+#include "ada/expected.h"
+#include <optional>
+#include <string_view>
+
+/**
+ * @namespace ada::parser
+ * @brief Code for href_aggregator
+ */
+namespace ada::parser {
+
+/**
+ * Sometimes, you do not want to construct a full URL structure and you would
+ * be satisfied with just a normalized href string.
+ *
+ * The href_aggregator contains a single string (buffer) along with a
+ * set of offsets (components). That is sufficient to qualified as a fully
+ * parsed URL. However, unlike ada/url, updates can be more difficult because
+ * the components are not separate. However, we should be able to map an
+ * ada/href_aggregator to an ada/url quickly and efficiently.
+ *
+ * In effect, we can view ada/url and ada/href_aggregator as equivalent.
+ * The ada/href_aggregator class is smaller and faster when processing essentially
+ * immutable URLs. The ada/url class is larger and has more overhead, but it would
+ * be well suited for URLs that are built by parts and modified. For many temporary
+ * or ephemeral instances, an href_aggregator would be more performant.
+ *
+ * TODO: both ada/url and ada/href_aggregator should have a common interface, 
```

Comment:
I think we should first define what the API is, and implementing the rest is quite easy. I believe we should leave the `setters` and `getters` of URL as it is, and just implement them according to the need of result_type. For now, if we extend `href_aggregator` from `ada::url` struct/class, we can override and keep the thing as it is and see where we differ from the original implementation. What do you think?

Reply:
This would mean that, when we have the single allocated buffer/string and implemented the rest, we could remove the inheritance, and remove the unnecessary declarations of `username` etc. in ada::url.